docs/sources/clients/promtail/stages/timestamp.md

```diff
@@ -33,6 +33,11 @@ timestamp:
   # be extracted or parsed. Valid values are: [skip, fudge].
   # Defaults to "fudge".
   [action_on_failure: <string>]
+
+  # Which action should be taken in case the timestamp is identical
+  # to or before the timestamp of a log line with the same labels.
+  # Valid values are: [skip, fudge]. Defaults to "fudge".
+  [action_on_duplicate: <string>]
```

Comment:
Defaulting to `fudge` feels backwards incompatible and surprising to me. As someone who cares about nanosecond precision, this would be a pretty big change that could break existing log analysis uses.

Reply:
That's a great point, defaulting to skip (same behaviour as today, loki throws an error) might be a better default

The reason I put fudge as default as the proposed solution is I would think most people care more about getting their logs into loki at all rather than it throwing an error and then not ingesting any more logs at all, at the potential cost of some accuracy.

Seeing the default for errored lines was fudge too it felt more like the "loki way" to put ingestion > accuracy.

That said, the argument of backwards compatibility is a really solid one, I'll change this

Reply:
Yeah, consistency and backwards compatibility often fight each other. I can see your point, just worrying it could cause problems downstream.

Reply:
changed the default to skip